kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/CustomResourceCrud1109Test.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes.client.mock;
+
+import io.fabric8.kubernetes.api.model.apiextensions.v1beta1.CustomResourceDefinition;
+import io.fabric8.kubernetes.api.model.apiextensions.v1beta1.CustomResourceDefinitionBuilder;
+import io.fabric8.kubernetes.client.dsl.MixedOperation;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.fabric8.kubernetes.client.dsl.base.CustomResourceDefinitionContext;
+import io.fabric8.kubernetes.client.mock.crd.FooBar;
+import io.fabric8.kubernetes.client.mock.crd.FooBarList;
+import io.fabric8.kubernetes.client.server.mock.KubernetesServer;
+import org.junit.Rule;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.migrationsupport.rules.EnableRuleMigrationSupport;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+@EnableRuleMigrationSupport
+class CustomResourceCrud1109Test {
+  @Rule
+  public KubernetesServer server = new KubernetesServer(true,true);
+
+  private CustomResourceDefinition customResourceDefinition;
+
+  @BeforeEach
+  void setUp() {
+    customResourceDefinition = server.getClient().apiextensions().v1beta1().customResourceDefinitions()
+      .create(new CustomResourceDefinitionBuilder()
+      .withNewMetadata()
+      .withName("foo-bar.baz.example.com")
+      .endMetadata()
+      .withNewSpec()
+      .withGroup("baz.example.com")
+      .addNewVersion().withName("v1alpha1").endVersion()
+      .withScope("Namespaced")
+      .withNewNames()
+      .withKind("FooBar")
+      .withPlural("foo-bars")
+      .withSingular("foo-bar")
+      .endNames()
+      .endSpec()
+      .build());
+  }
+
+  @Test
+  @DisplayName("Fix for issue 1109, verifies resources with dashes can be retrieved")
+  void test1109() {
+    // Given
+    final MixedOperation<FooBar, FooBarList, Resource<FooBar>> fooBarClient =
+      server.getClient().customResources(CustomResourceDefinitionContext.fromCrd(customResourceDefinition), FooBar.class, FooBarList.class);
```

Comment:
What do you think about using the single argument `customResource` method? I think it's more readable and will avoid creation of `FooBarList`.
```java
final MixedOperation<FooBar, KubernetesResourceList<FooBar>, Resource<FooBar>> fooBarClient = server.getClient().customResources(FooBar.class);
```

Reply:
https://github.com/fabric8io/kubernetes-client/pull/2664#discussion_r542526653 + this comment:

#2665

Reply:
Once #2665 is merged we can revisit or improve this test (if necessary). Right now its main use is just making sure 1109 doesn't happen again, especially considering what went on in  #2658 discussion about removing certain components.

Reply:
I've approved the PR already since these comments are really minor nitpicks.